main.js

```diff
@@ -620,6 +662,256 @@ app.post('/delete-admin', async function(req, res) {
+			return;
+		}
+		try {
+			const userFilterQuery = {'_id': new mdb.ObjectId(req.session._id)};
+			kitTemplate.kit = kitTemplate.kit.map(element => {
+				element.required = true;
+				element.completed = false;	
+				return element;
+			})
+			const user = await db.collection('BBY-6_users').updateOne(userFilterQuery, {$push: {kits: kitTemplate}});
+			if(!user) {
+				// Could not find user
+				res.status(500).send("userNotFound");
+				return;
+			}
+			res.status(200).send(kitTemplate);
+		} catch(e) {
+			res.status(500).send("serverIssue");
+		}
+	} catch(e) {
+		res.status(500).send("serverIssue");
+	}
+})
+
+app.patch('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	if(!req.body || !req.body._id || !req.body.itemId || req.body.completed === undefined) {
+		res.status(400).send("missingBodyArguments");
+		return;
+	}
+	const filterQuery = {
+		'_id': new mdb.ObjectId(req.session._id),
+		'kits._id': new mdb.ObjectId(req.body._id),
+	}
+	try {
+		const result = await db.collection('BBY-6_users').updateOne(filterQuery, {$set: {"kits.$[i].kit.$[j].completed": req.body.completed}}, 
+			{
+				arrayFilters: [ 
+					{"i._id": new mdb.ObjectId(req.body._id)}, 
+					{"j._id": new mdb.ObjectId(req.body.itemId)}
+				]
+			}
+		);
+		if(!result) {
+			// Could not find user 
+			res.status(500).send("couldNotFindUser");
+			return;
+		} else if (result.matchedCount === 0) {
+			// Could not find kit
+			res.status(404).send("couldNotFindKit");
+			return;
+		}
+		res.status(200).send("kitUpdatedSuccessfully");
+	} catch(e) {
+		res.status(500).send("serverIssue");
+	}
+})
+
+app.delete('/kits', async (req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	if(!req.body || !req.body._id) {
+		res.status(400).send("missingBodyArguments");
+		return;
+	}
+	const filterQuery = {
+		'_id': new mdb.ObjectId(req.session._id),
+		'kits._id': new mdb.ObjectId(req.body._id)
+	}
+	try {
+		req.body._id = new mdb.ObjectId(req.body._id);
+		const result = await db.collection('BBY-6_users').updateOne(filterQuery, {$pull: {"kits": {"_id": new mdb.ObjectId(req.body._id)}}});
+		if(!result) {
+			// Could not find user 
+			res.status(500).send("couldNotFindUser");
+			return;
+		} else if (result.matchedCount === 0) {
+			// Could not find kit
+			res.status(404).send("couldNotFindKit");
+			return;
+		}
+		res.status(200).send("kitDeletedSuccessfully");
+	} catch(e) {
+		console.log(e)
+		res.status(500).send("serverIssue");
+	}
+})
+
+app.post('/add-item', upload.single('image'), async(req, res) => {
+	if (redirectToLogin(req, res)) {
+		return;
+	}
+	if(!req.body || !req.body._id || !req.body.itemProps || !req.file) {
+		res.status(400).send("missingBodyArguments");
+		return;
+	}
+	const filterQuery = {
+		'_id': new mdb.ObjectId(req.session._id),
+		'kits._id': new mdb.ObjectId(req.body._id)
+	}
+	const itemProps = JSON.parse(req.body.itemProps);
+	const newItem = {
```

Comment:
No input sanitization, but it's fine. Only thing of concern *really* is an upload size limit on images. (But it's *fine.*)